File: microbenchmarks/cpp/integer_boolean/big_integer/LessEqual.cpp
#include <iostream>
#include <string>
#include "BUtils.cpp"
#include "BInteger.cpp"
#include "BBoolean.cpp"

#ifndef LessEqual_H
#define LessEqual_H

using namespace std;

class LessEqual {



    private:



        BInteger counter;

    public:

        LessEqual() {
            counter = (BInteger("0"));
        }

        void simulate() {
            while((counter.less((BInteger("5000000")))._and((BInteger("1")).lessEqual((BInteger("2"))))).booleanValue()) {
                counter = counter.plus((BInteger("1")));
            }
        }

};
int main() {
    clock_t start,finish;
    double time;
    LessEqual exec;
    start = clock();
    exec.simulate();
    finish = clock();
    time = (double(finish)-double(start))/CLOCKS_PER_SEC;
    printf("%f\n", time);
    return 0;
}
#endif

File: benchmarks/java/integer_boolean/Java_Primitive_Integer/LessEqual.java
import de.hhu.stups.btypes.BInteger;
import de.hhu.stups.btypes.BBoolean;
import de.hhu.stups.btypes.BUtils;

public class LessEqual {






    private BInteger counter;

    public LessEqual() {
        counter = (BInteger) new BInteger(0);
    }

    public void simulate() {
        while((counter.less(new BInteger(5000000)).and(new BInteger(1).lessEqual(new BInteger(2)))).booleanValue()) {
            counter = (BInteger) counter.plus(new BInteger(1));
        }
    }

    public static void main(String[] args) {
        LessEqual exec = new LessEqual();
        long start = System.nanoTime();
        exec.simulate();
        long end = System.nanoTime();
        System.out.println(exec.getClass().toString() + " Execution: " + (end - start));
    }

}
